service/history/timerQueueActiveProcessor.go

```diff
@@ -194,80 +194,45 @@ func (t *timerQueueActiveProcessorImpl) notifyNewTimers(timerTasks []persistence
 	t.timerQueueProcessorBase.notifyNewTimers(timerTasks)
 }
 
-func (t *timerQueueActiveProcessorImpl) process(timerTask *persistence.TimerTaskInfo) error {
+func (t *timerQueueActiveProcessorImpl) process(timerTask *persistence.TimerTaskInfo) (int, error) {
 	ok, err := t.timerTaskFilter(timerTask)
 	if err != nil {
-		return err
+		return metrics.TimerActiveQueueProcessorScope, err
 	} else if !ok {
 		t.timerQueueAckMgr.completeTimerTask(timerTask)
 		t.logger.Debugf("Discarding timer: (%v, %v), for WorkflowID: %v, RunID: %v, Type: %v, EventID: %v, Error: %v",
 			timerTask.TaskID, timerTask.VisibilityTimestamp, timerTask.WorkflowID, timerTask.RunID, timerTask.TaskType, timerTask.EventID, err)
-		return nil
+		return metrics.TimerActiveQueueProcessorScope, nil
 	}
 
-	scope := metrics.TimerActiveQueueProcessorScope
 	switch timerTask.TaskType {
 	case persistence.TaskTypeUserTimer:
-		scope = metrics.TimerActiveTaskUserTimerScope
-		err = t.processExpiredUserTimer(timerTask)
+		return metrics.TimerActiveTaskUserTimerScope, t.processExpiredUserTimer(timerTask)
 
 	case persistence.TaskTypeActivityTimeout:
-		scope = metrics.TimerActiveTaskActivityTimeoutScope
-		err = t.processActivityTimeout(timerTask)
+		return metrics.TimerActiveTaskActivityTimeoutScope, t.processActivityTimeout(timerTask)
 
 	case persistence.TaskTypeDecisionTimeout:
-		scope = metrics.TimerActiveTaskDecisionTimeoutScope
-		err = t.processDecisionTimeout(timerTask)
+		return metrics.TimerActiveTaskDecisionTimeoutScope, t.processDecisionTimeout(timerTask)
 
 	case persistence.TaskTypeWorkflowTimeout:
-		scope = metrics.TimerActiveTaskWorkflowTimeoutScope
-		err = t.processWorkflowTimeout(timerTask)
+		return metrics.TimerActiveTaskWorkflowTimeoutScope, t.processWorkflowTimeout(timerTask)
 
 	case persistence.TaskTypeActivityRetryTimer:
-		scope = metrics.TimerActiveTaskActivityRetryTimerScope
-		err = t.processActivityRetryTimer(timerTask)
+		return metrics.TimerActiveTaskActivityRetryTimerScope, t.processActivityRetryTimer(timerTask)
 
 	case persistence.TaskTypeWorkflowRetryTimer:
-		scope = metrics.TimerActiveTaskWorkflowRetryTimerScope
-		err = t.processWorkflowRetryTimer(timerTask)
+		return metrics.TimerActiveTaskWorkflowRetryTimerScope, t.processWorkflowRetryTimer(timerTask)
 
 	case persistence.TaskTypeDeleteHistoryEvent:
-		scope = metrics.TimerActiveTaskDeleteHistoryEvent
-		err = t.timerQueueProcessorBase.processDeleteHistoryEvent(timerTask)
-	}
-
-	t.logger.Debugf("Processing timer: (%v, %v), for WorkflowID: %v, RunID: %v, Type: %v, EventID: %v, Error: %v",
-		timerTask.TaskID, timerTask.VisibilityTimestamp, timerTask.WorkflowID, timerTask.RunID, timerTask.TaskType, timerTask.EventID, err)
+		return metrics.TimerActiveTaskDeleteHistoryEvent, t.timerQueueProcessorBase.processDeleteHistoryEvent(timerTask)
```

Comment:
can we rename TimerActiveTaskDeleteHistoryEvent to TimerActiveTaskDeleteHistoryEventScope

Reply:
ok